installers/agent/release/agent.sh

```diff
@@ -105,21 +143,18 @@ export AGENT_STARTUP_ARGS
 export LOG_DIR
 export LOG_FILE
 
-CMD="$JAVA_HOME/bin/java -jar \"$AGENT_DIR/agent-bootstrapper.jar\" $GO_SERVER $GO_SERVER_PORT"
+RUN_CMD=("$(autoDetectJavaExecutable)" "-jar" "$AGENT_DIR/agent-bootstrapper.jar" "$GO_SERVER" "$GO_SERVER_PORT")
 
-echo "[`date`] Starting Go Agent Bootstrapper with command: $CMD" >>"$STDOUT_LOG_FILE"
-echo "[`date`] Starting Go Agent Bootstrapper in directory: $AGENT_WORK_DIR" >>"$STDOUT_LOG_FILE"
-echo "[`date`] AGENT_STARTUP_ARGS=$AGENT_STARTUP_ARGS" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] Starting Go Agent Bootstrapper with command: ${RUN_CMD[@]}" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] Starting Go Agent Bootstrapper in directory: $AGENT_WORK_DIR" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] AGENT_STARTUP_ARGS=$AGENT_STARTUP_ARGS" >>"$STDOUT_LOG_FILE"
 cd "$AGENT_WORK_DIR"
 
-if [ "$JAVA_HOME" == "" ]; then
-    echo "Please set JAVA_HOME to proceed."
-    exit 1
-fi
 
 if [ "$DAEMON" == "Y" ]; then
-    eval exec nohup "$CMD" >>"$STDOUT_LOG_FILE" 2>&1 &
+    "${RUN_CMD[*]}"  >> "$STDOUT_LOG_FILE" 2>&1 &
```

Comment:
Also, why no nohup? I think there's a difference between disown and nohup. IIRC, something about reading from stdin. We probably don't read from stdin, but if we do, the process will probably be killed. I think.